pkg/latte/intent_approve_login_link.go

```diff
@@ -0,0 +1,79 @@
+package latte
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentApproveLoginLink{})
+}
+
+var IntentApproveLoginLinkSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"additionalProperties": false,
+		"properties": {
+			"code": { "type": "string" }
+		},
+		"required": ["code"]
+	}
+`)
+
+type IntentApproveLoginLink struct {
+	Code string `json:"code"`
+}
+
+func (i *IntentApproveLoginLink) Kind() string {
+	return "latte.IntentApproveLoginLink"
+}
+
+func (i *IntentApproveLoginLink) JSONSchema() *validation.SimpleSchema {
+	return IntentApproveLoginLinkSchema
+}
+
+func (i *IntentApproveLoginLink) GetCode() string {
+	return i.Code
+}
+
+func (i *IntentApproveLoginLink) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	return []workflow.Input{}, nil
+}
+
+func (i *IntentApproveLoginLink) ReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow, input workflow.Input) (*workflow.Node, error) {
+	codeModel, err := deps.OTPCodeService.SetUserInputtedMagicLinkCode(i.GetCode())
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = deps.OTPCodeService.VerifyMagicLinkCode(i.GetCode(), true)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the web session and trigger the refresh event
+	webSessionProvider := deps.GlobalSessionServiceFactory.NewGlobalSessionService(
```

Comment:
Not sure if we should use webapp session in workflow. What is the expected method for workflow API consumer to handle magic link approval?

Reply:
This approval is actually more of a ping to the original session, which will actually consume the login link token.

The reason of moving webapp session refresh to workflow was to make it latte-specific. In the future if we have a different implementation of login link auto-approve, it won't affect this feature

Reply:
So this approval would store the code somewhere, and the custom UI may retrieve it through some API. Do we have this API at the moment?

As a developer of custom UI, how would I detect user has approved the login link in browser/app? Do I poll the get workflow API to see any changes, or workflow API would provide real-time notification on workflow changes?

Reply:
Seems current workflow does not plan to support real-time notification yet so will need to use polling for now?

Reply:
We will add a websocket endpoint for workflow, like the websocket endpoint for webapp.